pkg/clients/openshift/client.go

```diff
@@ -91,3 +94,57 @@ func (c *Client) GetJobLogs(ctx context.Context, name, namespace string) (string
 	// TODO: there may be a case where the first item isn't correct
 	return c.GetPodLogs(ctx, pods.Items[0].GetName(), namespace)
 }
+
+// WatchJob watches a job status and returns error, if any
+func (c *Client) WatchJob(ctx context.Context, namespace, jobname string) error {
+	clientSet, err := kubernetes.NewForConfig(c.GetConfig())
+	if err != nil {
+		return fmt.Errorf("failed to create kubernetes clientset: %w", err)
+	}
+	jobs, err := clientSet.BatchV1().Jobs(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed listing jobs: %w", err)
+	}
+	for _, job := range jobs.Items {
+		if job.Name != jobname {
+			continue
+		}
+		if job.Status.Succeeded > 0 {
+			return nil
+		}
+	}
+	watcher, err := clientSet.BatchV1().Jobs(namespace).Watch(ctx, metav1.ListOptions{
+		ResourceVersion: jobs.ResourceVersion,
+		FieldSelector:   "metadata.name=" + jobname,
+	})
+	if err != nil {
+		return fmt.Errorf("failed watching job: %w", err)
+	}
```

Comment:
We should be able to simplify things slightly by using `resources.Resources.List` directly and using this helper

https://github.com/kubernetes-sigs/e2e-framework/blob/332536f9ed7ed22fd069c5622b057a91ceeb1208/klient/k8s/resources/resources.go#L277

Reply:
Can you share the code you tried? The below code compiles

```
package main

import (
    "context"

    "github.com/openshift/osde2e-common/pkg/clients/openshift"
    batchv1 "k8s.io/api/batch/v1"
    "k8s.io/klog/v2"
)

func main() {
    ctx := context.TODO()

    oc, err := openshift.New(klog.NewKlogr())
    if err != nil {
        panic(err)
    }

    jobs := new(batchv1.JobList)
    if err := oc.List(ctx, jobs); err != nil {
        panic(err)
    }
}
```

Reply:
There is still the `Watch` helper that we could use, it would cut down a bit on the event handling code but we can follow up with that.

https://github.com/kubernetes-sigs/e2e-framework/blob/332536f9ed7ed22fd069c5622b057a91ceeb1208/klient/k8s/resources/resources.go#L277